### PR TITLE
2189: Mark command as handled when JCheckConfiguration is missing or invalid in PullRequestCommandWorkItem

### PR DESCRIPTION
**Pull request by zhaosongzs**
Recently, we found that the skara bot keeps trying to process /open command in this pr https://github.com/openjdk/jdk8u-dev/pull/393

After investigation, we found that this pr is a dependent pr and this pr was closed before the parent pr(https://github.com/openjdk/jdk8u-dev/pull/392) got integrated, so this dependent pr didn't get retargeted.

Skara bot was trying to process the command but the bot couldn't find jcheck configuration in the target branch because the target is already deleted. And when the jcheck configuration is invalid or missing in PullRequestCommandWorkItem, RuntimeException would be thrown, causing the bot to repeatedly process the command. This behavior is wrong because when jcheck configuration is missing or invalid, it wouldn't recover automatically. In this case, the bot should just notice the user what problem it found and mark the command as handled to prevent the bot from being stuck in a loop.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace

### Issue
 * [SKARA-2189](https://bugs.openjdk.org/browse/SKARA-2189): Mark command as handled when JCheckConfiguration is missing or invalid in PullRequestCommandWorkItem (**Bug** - P4)

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/skara.git pull/1616/head:pull/1616` \
`$ git checkout pull/1616`

Update a local copy of the PR: \
`$ git checkout pull/1616` \
`$ git pull https://git.openjdk.org/skara.git pull/1616/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1616`

View PR using the GUI difftool: \
`$ git pr show -t 1616`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/skara/pull/1616.diff">https://git.openjdk.org/skara/pull/1616.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/skara/pull/1616#issuecomment-1974074771)

### CODE REVIEW

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/PullRequestCommandWorkItem.java [hunk 1]
```diff
@@ -200,13 +200,36 @@ public Collection<WorkItem> prRun(ScratchArea scratchArea) {
...
+                if (branchNames.contains(pr.targetRef())) {
+                    printer.println("JCheck configuration is missing or invalid in the target branch of this pull request.");
+                } else {
+                    printer.println("The target branch of this pull request no longer exists. Please retarget this pull request.");
+                }
+            } else {
+                log.severe(bot.confOverrideName() + " on " + bot.confOverrideRef() +
+                        " is missing or invalid in repo " + bot.confOverrideRepository().get().name());
+                printer.println("The JCheck configuration has been overridden, " +
+                        "but it's missing or invalid. Skara admin has been noticed and will fix it as soon as possible.");
```
Comment (erikj79) [inline bots/pr/src/main/java/org/openjdk/skara/bots/pr/PullRequestCommandWorkItem.java:227]:
```suggestion
                        "but is missing or invalid. Skara admins have been notified.");
```

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/PullRequestCommandWorkItem.java [hunk 2]
```diff
@@ -200,13 +200,36 @@ public Collection<WorkItem> prRun(ScratchArea scratchArea) {
...
+            var printer = new PrintWriter(writer);
+
+            printer.println(String.format(COMMAND_REPLY_MARKER, command.id()));
+            printer.print("@");
+            printer.print(command.user().username());
+            printer.print(" ");
+            if (bot.confOverrideRepository().isEmpty()) {
+                var branchNames = pr.repository().branches().stream().map(HostedBranch::name).toList();
+                if (branchNames.contains(pr.targetRef())) {
+                    printer.println("JCheck configuration is missing or invalid in the target branch of this pull request.");
```
Comment (erikj79) [inline bots/pr/src/main/java/org/openjdk/skara/bots/pr/PullRequestCommandWorkItem.java:219]: We do know if it's missing or invalid based on the type of the exception. I think it's worth differentiating the message based on this.

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/PullRequestCommandWorkItem.java [hunk 3]
```diff
@@ -200,13 +200,36 @@ public Collection<WorkItem> prRun(ScratchArea scratchArea) {
...
+                } else {
+                    printer.println("The target branch of this pull request no longer exists. Please retarget this pull request.");
+                }
+            } else {
+                log.severe(bot.confOverrideName() + " on " + bot.confOverrideRef() +
+                        " is missing or invalid in repo " + bot.confOverrideRepository().get().name());
+                printer.println("The JCheck configuration has been overridden, " +
+                        "but it's missing or invalid. Skara admin has been noticed and will fix it as soon as possible.");
+            }
+            printer.print("Please issue this command again once the problem has been resolved.");
```
Comment (erikj79) [inline bots/pr/src/main/java/org/openjdk/skara/bots/pr/PullRequestCommandWorkItem.java:229]: I don't think this should be printed on a separate line.

**Comment by zhaosongzs** [comment]
Thanks for the review!
/integrate

**Pull request closed without merging**